unlock-app/src/components/creator/members/GrantKeysDrawer.tsx

```diff
@@ -0,0 +1,316 @@
+ * so we format it for it to be used there.
+ * @param date
+ * @returns
+ */
+const formatDate = (date: Date) => {
+  date.setMinutes(date.getMinutes() - date.getTimezoneOffset())
+  return date.toISOString().slice(0, 16)
+}
+
+/**
+ * Form part
+ * @returns
+ */
+const GrantKeyForm = ({ onGranted, lock }: GrantKeyFormProps) => {
+  const { account, network } = useContext(AuthenticationContext)
+  const { openAlert, alertProps } = useAlert()
+
+  const walletService = useContext(WalletServiceContext)
+  const web3Service = useContext(Web3ServiceContext)
+
+  const [transaction, setTransaction] = useState<string>('')
+  const [loading, setLoading] = useState(false)
+
+  const defaultValues = {
+    recipient: '',
+    expiration: formatDate(new Date(now + lock.expirationDuration * 1000)),
+    keyManager: '',
+  }
+
+  const {
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors },
+    setValue,
+  } = useForm({
+    mode: 'onSubmit',
+    reValidateMode: 'onSubmit',
+    defaultValues,
+  })
+
+  useEffect(() => reset(defaultValues), [lock.name])
+
+  interface onSubmitInterface {
+    recipient: string
+    keyManager: string
+    expiration: string
+  }
+
+  const onSubmit = async ({
+    recipient,
+    keyManager,
+    expiration,
+  }: onSubmitInterface) => {
+    setLoading(true)
+    try {
+      const existingExpiration =
+        await web3Service.getKeyExpirationByLockForOwner(
+          lock.address,
+          recipient,
+          network
+        )
+      if (existingExpiration > new Date().getTime() / 1000) {
+        openAlert({
+          title: 'Error',
+          body: 'This address already owns a valid key. You cannot grant them a new one.',
+        })
+        onGranted(false)
+      } else {
+        await walletService.grantKey(
+          {
+            lockAddress: lock.address,
+            recipient,
+            expiration: Math.floor(new Date(expiration).getTime() / 1000),
+            keyManager: keyManager || account,
+          },
+          (error: any, hash: string) => {
+            if (error) {
+              console.error(error)
+              openAlert({
+                title: 'Error',
+                body: 'There was an error and the key could not be granted. Please refresh the page and try again.',
+              })
+            }
+            if (hash) {
+              setTransaction(hash)
+            }
+          }
+        )
+        setTransaction('')
+      }
+      onGranted(true)
+    } catch (error) {
+      console.error(error)
+      openAlert({
+        title: 'Error',
+        body: 'There was an error and the transaction could not be sent. Please refresh the page and try again.',
+      })
+      setTransaction('')
+    }
+    setLoading(false)
+  }
+
+  const addressFieldChanged = (name: string) => {
+    return async (event: any) => {
+      const address = await getAddressForName(event.target.value)
+      if (address) {
+        // @ts-expect-error
```

Comment:
Why the ts-expect error?

Reply:
Because it complains about the type :/ 

```
Argument of type 'string' is not assignable to parameter of type '"recipient" | "expiration" | "keyManager"'.ts
```
I am not sure what the best way but I fixed it with 

```
return setValue(
          name as 'recipient' | 'expiration' | 'keyManager',
          address,
          {
            shouldValidate: true,
            shouldDirty: true,
          }
        )
```